tools/src/main/scala/scala/scalanative/linker/Assembly.scala

```diff
@@ -16,14 +14,16 @@ object Assembly {
   final case class Dir private[Assembly](val base: File) extends Assembly {
     //println(s"discovered dir assembly $base")
 
+    def matchFiles(f: File): Array[File] = {
+      if (f.isDirectory) f.listFiles.map(matchFiles).flatten.toArray
+      else if (f.getName.endsWith(".nir")) Array(f)
+      else Array()
+    }
```

Comment:
Not sure if using arrays is a good idea for recursive construction of a sequence. Lists would probably be a better fit here.